internal/elasticsearch/index/ilm.go

```diff
@@ -568,15 +573,16 @@ var ilmActionSettingOptions = map[string]struct {
 	def            interface{}
 	minVersion     *version.Version
 }{
+	"allow_write_after_shrink": {def: false, minVersion: version.Must(version.NewVersion("8.14.0"))},
 	"number_of_replicas":       {skipEmptyCheck: true},
-	"total_shards_per_node":    {skipEmptyCheck: true, def: -1, minVersion: version.Must(version.NewVersion("7.16.0"))},
 	"priority":                 {skipEmptyCheck: true},
+	"max_primary_shard_docs":   {minVersion: version.Must(version.NewVersion("8.2.0"))},
```

Comment:
I am not sure about what should be put there as this property is optional. Maybe a skipEmptyCheck should be added there.

Reply:
I think this should be fine, `skipEmptyCheck` is around dealing with how the TF SDK represents unset attributes as Golang zero values.